server/src/services/initialization/onInitialized.ts

```diff
@@ -33,23 +34,24 @@ export const onInitialized = (
       });
     }
 
+    // index folders
     await serverState.telemetry.trackTiming("indexing", async () => {
       await indexWorkspaceFolders(
         { ...serverState, workspaceFolders: [] },
         workspaceFileRetriever,
         serverState.workspaceFolders
       );
+      serverState.indexingFinished = true;
```

Comment:
What do you think about moving indexing to initialize. Solving this the same way we solve the `validator still staring` error?

Reply:
Agreed.